lib/secrets.rb

```diff
@@ -65,6 +71,25 @@ def http_post(url, ssl_verify:, pem:)
     response.body
   end
 
+  def hostip
+    token = File.read(@serviceaccount_dir + '/token')
+    namespace = File.read(@serviceaccount_dir + '/namespace')
+    uri = URI.parse(@api_url + "/api/v1/namespaces/#{namespace}/pods")
+    req = Net::HTTP::Get.new(uri.path)
+    req.add_field("Authorization", "Bearer #{token}")
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = (uri.scheme == 'https')
```

Comment:
extract into a `http_get` method like the http_post method so this funky logic can stay in it's own little corner